src/main.rs

```diff
@@ -113,7 +113,7 @@ fn main() {
             Err(e) => {
                 eprintln!("failed to handle line");
                 eprintln!("  {}", DecodeHybrid::decode(&buffer));
-                eprintln!("  {}", e);
+                eprintln!("  {:?}", e);
```

Comment:
This is where implementing `Display` on `Error` would come in handy: you'd get a shiny error message, and this line wouldn't have to change.